core/src/trezor/ui/layouts/tr/__init__.py

```diff
@@ -1168,12 +1100,10 @@ def request_passphrase_on_host() -> None:
     )
 
 
-async def request_passphrase_on_device(ctx: GenericContext, max_len: int) -> str:
-    await button_request(
-        ctx, "passphrase_device", code=ButtonRequestType.PassphraseEntry
-    )
+async def request_passphrase_on_device(max_len: int) -> str:
+    await button_request("passphrase_device", code=ButtonRequestType.PassphraseEntry)
 
-    result = await ctx.wait(
+    result = await ctx_wait(
```

Comment:
Not sure why we do not call `interact` here (instead of manually calling `button_request` and then `ctx_wait`).

Seems like my fault, probably we were calling the result many times and then changed it to just once.

Reply:
81305badef3cbadaa7de0b696ce83953dae4089f